scripts/sscm.py

```diff
@@ -0,0 +1,773 @@
+            $ /path/to/scylla.git/scripts/{app_name} create-cluster -n2 -a
+            $ ./{app_name} status
+            [
+                {{
+                    "index": 1,
+                    "ip": "127.0.150.1",
+                    "status": "UP",
+                    "pid": 254315
+                }},
+                {{
+                    "index": 2,
+                    "ip": "127.0.150.2",
+                    "status": "UP",
+                    "pid": 254323
+                }}
+            ]
+            # Stop all nodes and stop the supervisor
+            $ ./{app_name} stop-cluster
+            """))
+    parser.add_argument("-c", "--cluster-directory", action="store", default=".",
+            help=f"the working directory of the cluster (must be writeable by {app_name})")
+    subparsers = parser.add_subparsers(dest="operation", help="the cluster/node operation to execute")
+
+    parser_create_cluster = subparsers.add_parser(
+            "create-cluster",
+            formatter_class=argparse.RawDescriptionHelpFormatter,
+            description=textwrap.dedent(f"""\
+            Setup a new cluster
+
+            Creates the required infrastructure, then starts the the supervisor.
+            Also creates a local symlink to {app_name} for convenience.
+            Should be run in an empty directory, but this is not enforced.
+
+            The following files are created:
+            * {app_name} - symlink to {app_name} (for convenience)
+            * {LOG_FILE} - supervisor logs here
+            * {PID_FILE} - contains the pid of the supervisor
+            * {FIFO_IN} - input fifo of the supervisor
+            * {FIFO_OUT} - output fifo of the supervisor
+            * {CONFIG_FILE} - containing the cluster's configuration
+
+            If the cluster is created with more than 0 nodes, there will be a directory
+            created for each, see add-node. The supervisor (and all the nodes) can be stopped
+            with stop-cluster. If there is already a cluster setup, you can use start-cluster
+            to start the supervisor.
+
+            The provided scylla-related parameters will serve as the defaults when adding new
+            nodes. They can be overriden on a node-by-node basis in add-node.
+            """),
+            help="create a new cluster")
+    parser_create_cluster.add_argument("--name", action="store", default="sccm-test-cluster",
+            help=f"name of the new cluster (will be passed to scylla's --cluster-name)")
+    parser_create_cluster.add_argument("-n", "--nodes", type=int, action="store", default=0,
+            help="amount of nodes to create the cluster with, the nodes are not started")
+    parser_create_cluster.add_argument("--network-prefix", type=int, action="store", default=None,
+            help="ip addresses for nodes will be constructed as 127.0.$network_prefix.X, if not provided, a random number will be used")
```

Comment:
Suggest to use 127.$prefix.$datacenter+$rack.$node, but of course it can be done later by whoever needs it.

Reply:
I will move the $prefix to the left to allow more seamless addition for this later.

Reply:
Done in v2.